### PR TITLE
Add type hints for several pytorch models (batch-4)

### PR DESCRIPTION
**Pull request by nablabits**
# What does this PR do?
Addresses some of the models in https://github.com/huggingface/transformers/issues/16059:
1. Mgpstr
	1. `MgpstrForSceneTextRecognition`
	2. `MgpstrModel`
2. PLBart
	1. `PLBartForConditionalGeneration`
	2. `PLBartModel`
3. `Pix2StructTextModel`
4. `RagModel`
5. `SamModel`
6. `Swin2SRModel`

## Who can review?

@Rocketknight1, please

### CODE REVIEW

**Comment by ArthurZucker** [comment]
cc @Rocketknight1

File: src/transformers/models/pix2struct/modeling_pix2struct.py [hunk 1]
```diff
@@ -1387,21 +1387,21 @@ def set_output_embeddings(self, new_embeddings):
...
-        cross_attn_head_mask=None,
-        past_key_values=None,
-        use_cache=None,
-        output_attentions=None,
-        output_hidden_states=None,
-        labels=None,
-        return_dict=None,
+        input_ids: Optional[torch.LongTensor] = None,
+        attention_mask: Optional[torch.FloatTensor] = None,
+        encoder_hidden_states: Optional[torch.Tensor] = None,
```
Comment (Rocketknight1) [inline src/transformers/models/pix2struct/modeling_pix2struct.py:1392]:
```suggestion
        encoder_hidden_states: Optional[torch.FloatTensor] = None,
```

File: src/transformers/models/rag/modeling_rag.py [hunk 2]
```diff
@@ -462,16 +462,12 @@ def from_pretrained_question_encoder_generator(
...
-
-            If the model has is not initialized with a `retriever` ``context_input_ids` has to be provided to the
-            forward pass. `context_input_ids` are returned by [`~RagRetriever.__call__`]. context_attention_mask
-            (`torch.LongTensor` of shape `(batch_size * config.n_docs, config.max_combined_length)`, *optional*,
-            returned when *output_retrieved=True*): Attention mask post-processed from the retrieved documents and the
-            question encoder `input_ids` by the retriever.
-
-            If the model has is not initialized with a `retriever` `context_attention_mask` has to be provided to the
-            forward pass. `context_attention_mask` are returned by [`~RagRetriever.__call__`].
+            retriever. If the model has is not initialized with a `retriever` ``context_input_ids` has to be provided
```
Comment (Rocketknight1) [inline src/transformers/models/rag/modeling_rag.py:465]:
```suggestion
            retriever. If the model was not initialized with a `retriever` ``context_input_ids` has to be provided
```

**Review by Rocketknight1 — APPROVED** [review]
Looks good to me! One quick type change, and one correction to the docstring - that bit wasn't your fault, but let's fix it while we're here.

File: src/transformers/models/pix2struct/modeling_pix2struct.py (hunk 1 above)
Comment (nablabits) [reply src/transformers/models/pix2struct/modeling_pix2struct.py]: Ahh, yes, I was unsure about this so I put the generic `Torch`. Thanks for pointing out.

**Pull request merged**